src/Shared/OpportunisticIntern.cs

```diff
@@ -21,116 +21,138 @@ namespace Microsoft.Build
     ///
     ///     string interned = OpportunisticIntern.Intern(String.Join(",",someStrings));
     ///
-    /// This class uses heuristics to decide whether it will be efficient to intern a string or not. There is no
+    /// There are currently two underlying implementations. The new default one in WeakStringCacheInterner is based on weak GC handles.
+    /// The legacy one in BucketedPrioritizedStringList is available only as an escape hatch by setting an environment variable.
+    ///
+    /// The legacy implementation uses heuristics to decide whether it will be efficient to intern a string or not. There is no
     /// guarantee that a string will intern.
     ///
     /// The thresholds and sizes were determined by experimentation to give the best number of bytes saved
     /// at reasonable elapsed time cost.
+    ///
+    /// The new implementation interns all strings but maintains only weak references so it doesn't keep the strings alive.
     /// </summary>
-    internal static class OpportunisticIntern
+    internal sealed class OpportunisticIntern
```

Comment:
Why sealed?

Reply:
If a class is not intended to have subclasses it is a good practice to seal it by default. Enables certain codegen optimizations - e.g. calling a virtual method can be faster because the runtime knows that the method cannot be overriden. It likely makes zero difference here but doing it anyway as a good practice.